torchdynamo/eval_frame.py

```diff
@@ -133,10 +136,32 @@ def __init__(self):
         super().__init__(callback=None)
 
 
+def my_compiler(gm: torch.fx.GraphModule, example_inputs):
+    print("inner my_compiler() called with FX graph:")
+    gm.graph.print_tabular()
+    return gm.forward
+
+
 def catch_errors_wrapper(callback):
     @functools.wraps(callback)
     def catch_errors(frame, cache_size):
         try:
+            # TODO(whc) move the ddp code below the bailouts. but make sure it doesn't cause DDP to be skipped
+            ddp_module = DistributedDataParallel.get_active_ddp_module()
+            if ddp_module and frame.f_code.co_name == "forward":
+                print(
+                    f"compile for ddp: {frame.f_code.co_name} {frame.f_code.co_filename}"
+                )
+                with compile_lock:
+                    ddp_optimizer = DDPOptimizer(
+                        bucket_bytes_cap=ddp_module.bucket_bytes_cap,
+                        parameters_to_ignore=ddp_module.parameters_to_ignore,
+                        backend_compile_fn=my_compiler,
```

Comment:
@anijain2305 @voznesenskym Any idea how I can get the  'my_compiler' function that the user provided in their 'torchdynamo.optimize' call?  I could probably figure out a way to hack it together, but I am hoping for a clean way.

Reply:
Not in front of it right now, I can check definitively later, but is that not `callback` in this func?

Reply:
originally i assumed `callback` was the thing, and i used it.  But i have been facing some (possibly unrelated) bugs, and when i read the code I thought callback is actually wrapped in some context which i am not sure i wanted to reuse.

Reply:
oh that might be `convert_frame` and friends. Try adding this line to convert_frame before the return

`_convert_frame._compiler_fn = compiler_fn`

And then see if your `callback` has a `_compiler_fn`? I can look in a moment as well.